src/compiler/checker.ts

```diff
@@ -7341,24 +7341,12 @@ namespace ts {
             unionIndex?: number;
         }
 
+        function getTypeId(type: Type) {
+            return type.id;
+        }
+
         function binarySearchTypes(types: Type[], type: Type): number {
-            let low = 0;

```

Comment:
we explicitly had two copies of this function to make sure this one is always optimized. we heavily rely on this function in the checker.

Reply:
Interestingly enough, this PR seems to have improved check time. I'll revert this specific change and re-run the benchmarks to see if there is any difference.

Reply:
The difference between the version in the PR and just reverting this single change is negligible (0% to 0.4% better after reverting the change). In most cases that's just within the standard error. I'd argue that manual inlining has only a minor overall impact and it might be worth the change to just have a single binary search algorithm.